bundles/org.eclipse.ui.workbench.texteditor/src/org/eclipse/ui/internal/findandreplace/overlay/FindReplaceOverlay.java

```diff
@@ -88,28 +91,36 @@ public class FindReplaceOverlay extends Dialog {
 
 	private Composite searchContainer;
 	private Composite searchBarContainer;
-	private Text searchBar;
+	private HistoryTextWrapper searchBar;
 	private AccessibleToolBar searchTools;
 	private ToolItem searchInSelectionButton;
 	private ToolItem wholeWordSearchButton;
 	private ToolItem caseSensitiveSearchButton;
 	private ToolItem regexSearchButton;
+
+	@SuppressWarnings("unused")
```

Comment:
general advice about @SuppressWarnings: As far as i understand those buttons are used in junit test via reflection. Either document that reason, or much better: do not use reflection. Instead for example offer and use getters. That makes code easier to maintain.

Reply:
This is a tricky one and has cause us quite some headaches before.

1) We don't want to expose the buttons to the outside world (at that point, we could just as well make them public)
2) The old tests used reflection - which is a bummer, but "works". We ended up using that method as well, knowing that it has problems
3) The bigger problem is that the tests are not in the same package as the Overlay: we cannot even make the members package-private since the test still wouldn't know about them
4) We tried looking into Reddeer and SWTBot for solving this issue. Unfortunately, neither are in the dependencies of Eclipse and we didn't want to introduce a new dependency (+ there were some other problems if I remember correctly)

I can add a comment to explain why I suppress warnings. I would opt for adding a single comment on the top one, I think that there's no need to repeat myself

Reply:
I have added a comment to explain what is happening and referred to this comment for future reference: https://github.com/eclipse-platform/eclipse.platform.ui/commit/060b9941fea88d89e98acb6b8adea55cde83bbe7

Reply:
You can make them package private and use a testcode in the same package. But since the class is "internal" only, so it is not a problem to make public methods.
On the other hand you probably just don't strongly need those Buttons as members but can find the widgets from the  swt tree.